core/controllers/suggestion.py

```diff
@@ -83,13 +89,65 @@ class SuggestionHandler(base.BaseHandler):
 
     @acl_decorators.can_suggest_changes
     def post(self):
+        change_dict = self.payload.get('change')
+        target_id = self.payload.get('target_id')
+        target_type = self.payload.get('target_type')
+        target_version_at_submission = self.payload.get(
+            'target_version_at_submission')
+        suggestion_type = self.payload.get('suggestion_type')
+        description = self.payload.get('description')
+
+        suggestion_domain_class = (
+            suggestion_registry.SUGGESTION_TYPES_TO_DOMAIN_CLASSES[
+                suggestion_type])
+        # While creating the below domain object fields suggestion_id and
+        # score_category is kept None because values for these fields are
+        # created in the create_suggestion() method. Also the domain object
+        # created below will only be used for extracting the HTML from the
+        # suggestions and these fields are not required for HTML extraction.
+        suggestion = suggestion_domain_class(
+            None, target_id, target_version_at_submission,
+            suggestion_models.STATUS_IN_REVIEW, self.user_id, None, change_dict,
+            None)
+        html_list = suggestion.get_all_html_content_strings()
+        filenames = (
+            html_cleaner.get_image_filenames_from_html_strings(html_list))
+        # In the case of edit state content suggestion the images are already
+        # saved to GCS and we don't need to save them in this controller.
+        if suggestion_type == (
+                suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT):
```

Comment:
For reviewers, I'm going to add `suggestion.get_target_entity_html_list_in_suggestion` which will return the HTML content which is part of the target entity and it's been used in the suggestion as well. Through that, I'll be able to exclude the filenames of the entity/exploration images.